bindings/rust-examples/async-pkey-offload/tests/client_server.rs

```diff
@@ -0,0 +1,106 @@
+use async_pkey_offload::{create_self_signed_cert, get_key, KmsAsymmetricKey, DEMO_DOMAIN, DEMO_REGION};
+use aws_config::{BehaviorVersion, Region};
+use aws_sdk_kms::Client;
+use s2n_tls::security;
+use s2n_tls_tokio::{TlsAcceptor, TlsConnector};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+    task::JoinHandle,
+};
+
+const MESSAGE: &[u8] = b"hello world";
+
+// we need multiple threads, because block_on can only be used in multi-threaded
+// runtimes
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn handshake() -> Result<(), Box<dyn std::error::Error>> {
+    let kms_key = {
+        let shared_config = aws_config::defaults(BehaviorVersion::v2024_03_28())
+            .region(Region::from_static(DEMO_REGION))
+            .load()
+            .await;
+        let kms_client = Client::new(&shared_config);
+        let key_id = get_key(&kms_client).await?;
+        println!("Using KMS Key: {:?}", key_id);
+        KmsAsymmetricKey::new(kms_client.clone(), key_id)
+            .await
+            .unwrap()
+    };
+
+    let self_signed_cert = create_self_signed_cert(kms_key.clone())?;
+    // closures are `move`, so we need another copy for the client.
```

Comment:
Nit: Are you using closures? It seems like the async move block is what's requiring the copy?